### PR TITLE
[GITHUB-1] Adds Jenkins Server role

### PR DESCRIPTION
**Pull request by lobsterdore**

### CODE REVIEW

**Comment by lobsterdore** [comment]
Ready for review.

**Review by rsgoodall — APPROVED** [review]

File: meta/main.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+---
+
+galaxy_info:
+  author: "David Reed"
```
Comment (racer1680) [inline meta/main.yml]: Do we need it?

File: tasks/configure.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  template:
+    dest: "{{ jenkins_server.directories.home }}/init.groovy.d/basic-security.groovy"
+    group: "{{ jenkins_server.group }}"
+    mode: 0600
+    owner: "{{ jenkins_server.user }}"
+    src: basic-security.groovy.j2
+  register: jenkins_service_admin_user
+  when: jenkins_service_init_groovy.changed
+
+- name: Ensure Jenkins is restarted if admin user created
```
Comment (racer1680) [inline tasks/configure.yml]: restarted or started?

Comment (lobsterdore) [reply tasks/configure.yml]: It's restarted so the user is created.

File: tasks/install.yml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+- name: Download Jenkins CLI from local server
+  become: yes
+  become_user: "{{ jenkins_server.user }}"
+  get_url:
+    dest: "{{ jenkins_server.directories.home }}"
+    url: "http://{{ jenkins_server.hostname_local }}:{{ jenkins_server.http_port }}/jnlpJars/jenkins-cli.jar"
+  when: not jenkins_service_cli.stat.exists
+
+# Use CLI as Ansible module seems to have issues with newer Jenkins versions
+- name: Install Jenkins plugins via CLI
```
Comment (racer1680) [inline tasks/install.yml]: Ansible does it with jenkins_plugin and jenkins 2.54

File: meta/main.yml (hunk 1 above)
Comment (lobsterdore) [reply meta/main.yml]: I think it's required for Galaxy, could change to a generic value of 'Sansible' maybe?

File: tasks/install.yml (hunk 3 above)
Comment (lobsterdore) [reply tasks/install.yml]: Doesn't work properly with newer Jenkins versions (as mentioned in the comment :) )

Comment (racer1680) [reply tasks/install.yml]: `jenkins_server:
  plugins:
    ansicolor:
      version: "0.4.3"
      enabled: yes`

File: defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    - "-Xmx{{ (ansible_memtotal_mb / 2) | int }}m"
+    - "-Djava.awt.headless=true"
+    - "-Dhttp.proxyHost=a-proxy-p.bc.jsplc.net"
+    - "-Dhttp.proxyPort=8080"
+    - "-Dhttps.proxyHost=a-proxy-p.bc.jsplc.net"
+    - "-Dhttps.proxyPort=8080"
+  java_opts_extra: [ ]
+  java_version: 8
+  plugins:
+    - ansicolor
```
Comment (racer1680) [inline defaults/main.yml]: where is version?

**Comment by racer1680** [comment]
LGTM

**Review by rsgoodall — APPROVED** [review]

**Review by rsgoodall — APPROVED** [review]
Looks fine.

**Review by rsgoodall — APPROVED** [review]

**Review by rsgoodall — APPROVED** [review]
Looks fine to me.

**Pull request merged**